steps/discover-repos.rb

```diff
@@ -69,17 +74,34 @@
   'NOT',
   'android'
 ].join(' ')
-loop do
-  if page * size > max
-    puts "Can't go to page ##{page}, since it will be over #{max}"
-    break
+
+def mock_array(size, licenses)
+  Array.new(size) do
+    {
+      full_name: "foo/#{Random.hex(5)}",
+      created_at: Time.now,
+      license: { key: licenses.sample(1)[0] }
+    }
   end
-  json = if opts[:dry]
-    { items: page > 100 ? [] : Array.new(size) { { full_name: "foo/#{Random.hex(5)}", created_at: Time.now } } }
+end
+
+def mock_reps(page, size, licenses)
+  {
+    items: if page > 100 then []
+    else
+      mock_array(size, licenses)
+    end
+  }
+end
+
+loop do
+  break if page * size > max
+  json = if opts[:dry] then mock_reps(page, size, licenses)
   else
     github.search_repositories(query, per_page: size, page: page)
   end
   json[:items].each do |i|
+    next if i[:license].nil? || !licenses.include?(i[:license][:key])
```

Comment:
@Orillio I suggest to print a log line, when this happens

Reply:
@yegor256 I wanted to add this logic, but this block is reached its maximum line length of 30.

I could have carry out this chunk of code to another function, but it wont allow me, because CI linter allows functions with length no more than 10 lines, which is highly unreasonable in my opinion.
```ruby
found[i[:full_name]] = {
  full_name: i[:full_name],
  default_branch: i[:default_branch],
  stars: i[:stargazers_count],
  forks: i[:forks_count],
  created_at: i[:created_at].iso8601,
  size: i[:size],
  open_issues_count: i[:open_issues_count],
  description: i[:description],
  topics: i[:topics]
}
```
At this point, i dont know how to add new lines here without fully refactoring the code

Reply:
@yegor256 I somehow managed to overcome the issue. Please check the commit.